### DISCUSSION TITLE
Source generator to automatically bind properties in custom controls

### DISCUSSION THREAD

**Discussion by camionkraken** [CommunityToolkit/Maui#530] [New Feature Discussions]
Hello,

I tried proposing my library in discussion #458 but I didn't receive an answer, so maybe I had to start a new discussion.
I'm sorry if I'm doing it the wrong way.

Anyway, my library (CCPropGen) aims to automatically create bindable properties for controls inside a `ContentView`. I provide an example to get the concept better.

## Example

Let's say that inside your `ContentView` there is an `Entry` and you want to bind its `Text` property.

My library exposes an attribute that, applied to the custom control class and given some required parameters, creates a bindable property linked to the `Text` property of the `Entry`.

Link to the repo: https://github.com/Riccardo11/CCPropGen.

This is a code example.

```csharp
using Maui.CCPropGen;

namespace CCPropGen.Maui.Demo;

// Attribute parameters:
// - Control name (set with x:Name in xaml)
// - Control type
// - Property name
// - Property type
[CCPropGen("EntryControl", typeof(Entry), "Text", typeof(string))]
public partial class CCPropGenEntry : ContentView
{
    public CCPropGenEntry()
    {
        InitializeComponent();
        
        // Currently needed to set binding contexts correctly
        InitializeCCPropGen();
    }
}
```

Now `CCPropGenEntry` has a `Text` property ready to be bound.

It's far from being ready to be used in a real environment, but I would appreciate your thoughts about it.

**Comment by camionkraken** [comment]
I'm adding some features whose could be interesting thinking about this library in a production environment, such as binding more properties from the same control or different controls.

I would really appreciate your opinion about this plugin.
Do you think it could be useful in the Community Toolkit?
Thank you.

**Reply by pictos** [reply]
Hey, @Riccardo11 thank you, I'll take a look at all implementations that the community shared with us and write a proposal soon-sh

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889
